ext/spl/spl_array.c

```diff
@@ -301,7 +286,7 @@ static zend_result get_hash_key(spl_hash_key *key, spl_array_object *intern, zva
 		ZVAL_DEREF(offset);
 		goto try_again;
 	default:
-		spl_array_illegal_offset(offset);
+		zend_illegal_container_offset("ArrayObject", offset, BP_VAR_R);
```

Comment:
`get_hash_key` is called for various purposes, so `R` will be inconsistent.

Reply:
So this is "funny" we actually throw the exception at call site of this function again, so the solution here is to just not throw the exception within the function.